src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -264,20 +264,64 @@
                :message (str "Invalid service_status_version. Should be an "
                           "integer but was " level)}))))
 
+(schema/defn ^:always-validate summarize-states
+  "Given a map of service statuses:
+   * if all of the statuses have the same :state, returns that :state
+   * if not all of the statuses are the same, returns :unknown"
+  [statuses :- ServicesStatus] :- State
+  (let [state-set (->> statuses
+                       vals
+                       (map :state)
+                       set)]
+    (if (= 1 (count state-set))
+      (first state-set)
+      :unknown)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Compojure App
 
-(defn build-routes
+(schema/defn ^:always-validate status->code
+  "Given a service status, returns an appropriate HTTP status code"
+  [status :- ServiceStatus] :- schema/Int
+  (if (nominal? status) 200 503))
+
+(schema/defn ^:always-validate statuses->code
+  "Given a map of service statuses, returns an appropriate HTTP status code."
+  [statuses :- ServicesStatus] :- schema/Int
+  (if (all-nominal? statuses)
+    200
+    503))
+
+(defn build-plaintext-routes
+  [path status-fns]
+  (comidi/context path
+     (comidi/GET "" _
+       (let [statuses (call-status-fns status-fns :critical)]
+         (ringutils/with-plain-response (statuses->code statuses)
+           (-> statuses
+               summarize-states
+               name))))
+
+     (comidi/GET ["/" :service-name] [service-name :as _]
```

Comment:
If this is functionally identical to compojure you don't have to bind the whole request w/ `:as _`; you can just leave that part out if you're not using it.